src/server.lisp

```diff
@@ -220,10 +220,21 @@ These warnings are included in the RPC response that is returned to the caller.
                                (finish-output *error-output*)
                                (trivial-backtrace:print-backtrace c :output *error-output*)))))
                  (apply f (concatenate 'list positional-args kwargs-as-plist)))))
-        (let ((result (if timeout
-                          (bt:with-timeout (timeout)
-                            (apply-handler))
-                          (apply-handler))))
+        (let* ((client-timeout (|RPCRequest-client_timeout| request))
+               ;; TODO A timeout of NIL is supposed to signal an
+               ;; indefinite timeout. The same is signalled by 0,
+               ;; which leads to the extra logic below and means
+               ;; subtle bugs may creep in.  Can we be strict about
+               ;; NIL vs 0?
```

Comment:
Should probably add a null check here for `timeout` then, since that is the default (and therefore probably most common) value. Otherwise, the `(> timeout client-timeout)` will signal a type error.